app/src/LightingManager.cpp

```diff
@@ -38,13 +69,22 @@ CHIP_ERROR LightingManager::Init()
     if (envGPIO == NULL || strlen(envGPIO) == 0)
     {
         ChipLogError(AppServer, "Environment variable not set or empty: %s", GPIO);
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
 
+    if (!convertStringToIntNoExcept(envGPIO, gpio))
+    {
+        ChipLogError(AppServer, "Non-integer value for GPIO: %s", envGPIO);
```

Comment:
-1, which is considered an error is not "Non-integer".

Reply:
-1 is considered as an Invalid value error

Reply:
I was commenting on the error message: "Non-integer value for GPIO: -1".
-1 is an integer but the error says it isn't.

Reply:
-1 is handled in the next block.
```
sudo snap set matter-pi-gpio-commander gpio=-1
sudo snap start matter-pi-gpio-commander
```
```
CHIP:SVR: Invalid value for GPIO: -1
```

Reply:
I see. Please forgive my ignorance.